# **CROP WISE AREA,PRODUCTION AND YIELD ANALYSIS ACROSS INDIA**

# Objective (Problem Statement)



1.  To identify crop production patterns, yield variations, and cultivation trends across different states, districts, crops, and seasons in India using exploratory data analysis (EDA).
2.  To clean, transform, and preprocess large-scale agricultural data for accurate and reliable analysis.
3.   To analyze the relationship between cultivated area, production, and yield for evaluating agricultural performance.
4.   To generate actionable insights and recommendations that support agricultural planning, resource allocation, and productivity improvement.









# Source

India Data Portal [(https://indiadataportal.com/p/area-production-yield-apy/r/moafw-area_production_yield-dt-sn-qzf)]

# Timeline

Multi-year agricultural records collected across India.

# Location

India

# Type of Problem

Exploratory Data Analysis (EDA)




# Dataset Features Description

| Feature Name | Description |
|-------------|-------------|
| year | Year of observation |
| state_name | State name |
| state_code | State code |
| district_name | District name |
| district_code | District code |
| crop_name | Crop cultivated |
| crop_type | Category of crop |
| season | Growing season |
| area | Cultivated area |
| area_unit | Unit of area measurement |
| production | Total production |
| production_unit | Unit of production |
| yield | Yield obtained |
| yield_unit | Unit of yield measurement |




In [2]:
##Initial EDA
##Data Loading and Initial Overview
##Import Required Libraries

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
##Load Dataset from GitHub

url="https://raw.githubusercontent.com/kaviraj2405-jpg/crop-wise-area-production-yield-analysis/refs/heads/main/crop-wise-area-production-yield.csv"

df = pd.read_csv(url)

print(df.shape)
df.head()

(455359, 16)


,id,year,state_name,state_code,district_name,district_code,crop_name,crop_code,crop_type,season,area,area_unit,production,production_unit,yield,yield_unit
0,0,1997-1998,Andhra Pradesh,28,Ananthapuramu,502,Arhar/Tur,202.0,Pulses,Kharif,21400.0,Hectare,2600.0,Tonnes,0.121,Tonnes/Hectare
1,1,1997-1998,Andhra Pradesh,28,Ananthapuramu,502,Bajra,103.0,Cereals,Kharif,1400.0,Hectare,500.0,Tonnes,0.357,Tonnes/Hectare
2,2,1997-1998,Andhra Pradesh,28,Ananthapuramu,502,Castor Seed,1002.0,Oilseeds,Kharif,1000.0,Hectare,100.0,Tonnes,0.100,Tonnes/Hectare
3,3,1997-1998,Andhra Pradesh,28,Ananthapuramu,502,Cotton(Lint),1101.0,Fiber Crops,Kharif,7300.0,Hectare,9400.0,Tonnes,1.288,Tonnes/Hectare
4,4,1997-1998,Andhra Pradesh,28,Ananthapuramu,502,Dry Chillies,502.0,Spices,Kharif,3700.0,Hectare,7100.0,Tonnes,1.919,Tonnes/Hectare


# Number of Rows and Columns

# Observation
Number of Rows: 455,359

Number of Columns: 16

In [3]:
#Data Types of Each Column
df.dtypes

,0
id,int64
year,object
state_name,object
state_code,int64
district_name,object
district_code,int64
crop_name,object
crop_code,float64
crop_type,object
season,object


#Data Type
#Observation

The dataset contains both numerical and categorical variables.


In [4]:
#Dataset Information
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 455359 entries, 0 to 455358
Data columns (total 16 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   id               455359 non-null  int64  
 1   year             455359 non-null  object 
 2   state_name       455359 non-null  object 
 3   state_code       455359 non-null  int64  
 4   district_name    455359 non-null  object 
 5   district_code    455359 non-null  int64  
 6   crop_name        455359 non-null  object 
 7   crop_code        455338 non-null  float64
 8   crop_type        455359 non-null  object 
 9   season           455359 non-null  object 
 10  area             455359 non-null  float64
 11  area_unit        455359 non-null  object 
 12  production       450350 non-null  float64
 13  production_unit  455359 non-null  object 
 14  yield            455359 non-null  float64
 15  yield_unit       455359 non-null  object 
dtypes: float64(4), int64(3), object(9)
mem

# Data Information

# Observation
*   Total entries
*   Column names
*   Data types
*   Non-null values
*   Memory usage










In [5]:
#Statistical Summary

df.describe()

,id,state_code,district_code,crop_code,area,production,yield
count,455359.000000,455359.000000,455359.000000,455338.000000,4.553590e+05,4.503500e+05,455359.00000
mean,227679.000000,19.412593,359.423745,468.182563,1.313002e+04,8.380320e+05,68.84521
std,131450.964951,9.430053,185.520710,368.573998,4.846655e+04,2.028612e+07,861.71475
min,0.000000,1.000000,1.000000,101.000000,4.000000e-03,0.000000e+00,0.00000
25%,113839.500000,10.000000,193.000000,107.000000,7.900000e+01,9.100000e+01,0.56300
50%,227679.000000,21.000000,375.000000,288.000000,6.000000e+02,8.080000e+02,1.05100
75%,341518.500000,28.000000,520.000000,708.000000,4.777000e+03,8.350000e+03,2.55000
max,455358.000000,38.000000,765.000000,1503.000000,8.580100e+06,1.597800e+09,43958.33300


# **Statistical Summary**

#Observation



* Mean
* Median
* Standard deviation
* Minimum value
* Maximum value
* Quartiles

In [6]:
#Finding Duplicates

df.duplicated().sum()

np.int64(0)

# **Finding Dupilcates**

# Observation

* No duplicates found

In [7]:
#Finding Missing Values

df.isnull().sum()

,0
id,0
year,0
state_name,0
state_code,0
district_name,0
district_code,0
crop_name,0
crop_code,21
crop_type,0
season,0


# **Finding Missing Values**

# Observation

* Missing values found in Crop_code
* 5009 Missing values found in Production

# **Target Feature**

# Observation

Not Applicable

This dataset is used for exploratory analysis and does not contain target variable.


# **Information About the Dataset**
* **Domain**: Agriculture
* **Dataset Type:** CSV
* **Total Records:** 455,359
* **Total Features:** 16
* **Numerical Features:** Area, Production, Yield, Year
* **Categorical Features:** State, District, Crop, Crop Type, Season
* **Project Category:** Exploratory Data Analysis (EDA)
* **Tools Used:** Python, Pandas, NumPy, Matplotlib, Seaborn